### ISSUE TITLE
Touchable doesn't recognise styled components as native components (react native)

### ISSUE THREAD

**Issue by dronsfield** [styled-components/styled-components#245]
## Version
1.1.1

## Description

Touchable components, such as TouchableHighlight, require their child to be a native component. An error is thrown when the child is a styled component. I don't think this is expected behaviour(?)

### Code example

```
import React from 'react';
import { AppRegistry, View, TouchableHighlight, Alert, } from 'react-native';
import styled from 'styled-components/native';

const Foo = styled.View`
	width: 100;
	height: 100;
	background-color: red;
`;

const App = (props) => (
	<TouchableHighlight onPress = { () => Alert.alert("bar") } >
		<Foo/>
	</TouchableHighlight>
);

export default App;
AppRegistry.registerComponent('myApp', () => App);
```

Running this code in-app throws the following error:

```
Touchable child must either be native or forward setNativeProps to a native component

handleException @ ExceptionsManager.js:62
handleError @ InitializeJavaScriptAppEngine.js:120
reportFatalError @ error-guard.js:30guard @ MessageQueue.js:47
callFunctionReturnFlushedQueue @ MessageQueue.js:104
onmessage @ debuggerWorker.js:44
```

### Similar example but without styled components, doesn't throw any errors

```
import React from 'react';
import { AppRegistry, View, TouchableHighlight, Alert, } from 'react-native';
import styled from 'styled-components/native';

const Foo = View;

const style = {
	width: 100,
	height: 100,
	backgroundColor: "red",
}

const App = (props) => (
	<TouchableHighlight onPress = { () => Alert.alert("bar") } >
		<Foo style = { style } />
	</TouchableHighlight>
);

export default App;
AppRegistry.registerComponent('myApp', () => App);
```

**Comment by mxstbr** [comment]
Definitely a bug, will try to get it fixed asap. I wonder why `setNativeProps` isn't being passed on! @geelen, any ideas?

**Comment by jh3y** [comment]
As a quick work around before a fix @dronsfield, if you wrap your children in a `View` you shouldn't get any errors 👍

**Comment by stief510** [comment]
+1

**Comment by stief510** [comment]
This is still not working for me with version 1.4.0

**Comment by yordis** [comment]
@mxstbr this still being a issue

```jsx

const Button = styled.TouchableHighlight`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 40;
  height: 40;
  border-radius: 50;
  background-color: ${props => props.disabled ? 'hsl(0, 0%, 95%)' : COLOR};
`

const StyledIcon = styled(Icon)`
  color: ${props => props.disabled ? 'hsla(0, 0%, 70%, 1)' : 'white'};  
`

<Button onPress={onSendMessage} disabled={!canSubmitMessage}>
    <StyledIcon name="paper-plane" disabled={!canSubmitMessage}/>
</Button>
```

Version: `"styled-components": "^2.2.4",`

I had to wrap `StyledIcon` on an `View` as @jh3y said

**Comment by philpl** [comment]
@yordis can you open an issue with a reproduction please? I suspect it’s a usage problem. This has been fixed in v2 again a longer time ago and this issue is stale now

**Comment by yordis** [comment]
@philpl before I open an new issue here is the full code

```jsx
// chat-input.js

import React from 'react'
import Icon from 'react-native-fa-icons'
import styled from 'styled-components/native'
import { View } from 'react-native'
import { COLOR } from './constants'

const InputContainer = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: white;
 
  height: 50;
  
  padding-left: 10;
  padding-right: 10;
 
  shadow-color: hsla(0, 0%, 10%, 0.15);
  shadow-offset: 0px 0px;
  shadow-opacity: 1;
  shadow-radius: 20;
`

const Input = styled.TextInput`
  flex: 1;
`

const Button = styled.TouchableHighlight`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 40;
  height: 40;
  border-radius: 50;
  background-color: ${props => props.disabled ? 'hsl(0, 0%, 95%)' : COLOR};
`

const StyledIcon = styled(Icon)`
  color: ${props => props.disabled ? 'hsla(0, 0%, 70%, 1)' : 'white'};  
`

export const ChatInput = (props) => {
  const { value, onChange, onSendMessage, canSubmitMessage } = props

  return (
    <InputContainer>
       <Input
         value={value}
         onChangeText={onChange}
         onSubmitEditing={onSendMessage}
         placeholder="Type message here"
       />
       <Button onPress={onSendMessage} disabled={!canSubmitMessage}>
         {/*<View>*/}
           <StyledIcon name="paper-plane" disabled={!canSubmitMessage}/>
         {/*</View>*/}
       </Button>
     </InputContainer>
   )
}


// chat.js who use it

<ChatInput
            value={currentMessage}
            onChange={this.setCurrentMessage}
            canSubmitMessage={this.canSubmitMessage()}
            onSendMessage={this.sendMessage}
          />

```

@philpl here is the catch, the component is rendering fine the first time but `when I click on it is when this happens`.

**Comment by philpl** [comment]
@yordis React-Native-fa-icons is actually missing setNativeProps. Please open a new issue over there so they can fix it.

Btw the reason why it’s counterproductive to comment on old / stale / closed issues is that you might have a usage problem, or an entirely different problem / bug, and at the same time you’re pinging people who are not interested in this issue anymore :)

You can also always open a thread on Spectrum where it’s pretty easy to help you while it’s still possible for us to keep track of all questions

**Comment by yordis** [comment]
@philpl here is the issue, `I don't know what I don't know`, if I open an new issue, then you complain because it is a usage issue (How I suppose to know that, I wish I am perfect and have unlimited knowledge but I don't and I will never have).
I am pinging the core contributor of the library who was the one that closed the issue, which for me is the one who probably understands why is was closed.

Also, one of the main reason why people should actually keep most of the activity on Github, which help people surfing the open and closed tickets (everything here is related to styled-components)

Spectrum is a separate community, as StackOverflow is, I don't mind to use it but understand that Github is the main platform for the code. Who `ensure` that Spectrum will no go away tomorrow and with it all the data?

**Comment by yordis** [comment]
@philpl also I notice that the only time you mention Spectrum is the template of the issue. I recommend to add some section on the readme where you suggest people to use Spectrum as the main platform for get help.

**Comment by philpl** [comment]
@yordis I promise you I wouldn’t complain on any new issue :) so feel free to ask usage questions in new issues if you can’t find the answer elsewhere

**Comment by yordis** [comment]
@philpl from core contributor on https://github.com/styled-components/styled-components/issues/1340#issuecomment-348705480
> Next time when you have a usage question please ask in the community or on StackOverflow, this is an issue tracker. Thanks!

So basically I shouldn't be using issues for that, soo what should I do ***(which I totally blame Github for all these issues)*** we need an stackoverflow on Github for multiple reasons.

But he missed the point of `I don't know what I don't know` so for him it was simple to understand what was the issue but his first response didn't help 😢 I was like: `what? I don't get what is the issue` https://github.com/styled-components/styled-components/issues/1340#issuecomment-348701166 so basically he got mad at me because of my ignorance  (sad because I am a beginner).

Anyway, I will use Spectrum for all the new questions and hopefully everything goes fine ❤️

**Comment by philpl** [comment]
@yordis I’d be happy if you could open an issue describing your trouble with our current system, so we can find good alternatives 👍

We prefer it if people opened issues for something other than usage questions, since then all issues go to members of our community as well.

The sad truth is that our time is very limited right now, meaning we don’t have time to answer quickly to questions on GitHub. We’re trying our best to help out, and I apologise if it came across as rude, but our intention is just to quickly resolve issues into github and to hope that people move them to one of our other channels since more people have more time to help you there 😊

Also feel free to ping me in DMs on Spectrum/Twitter if you have any beginner questions that need some short & quick answers

**Comment by sibelius** [comment]
sorry to reopen this

but this is failing for me in latest react-native version v62 with styled-components v5
